Tests/Builder/DatagridBuilderTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DoctrineORMAdminBundle\Tests\Builder;
+
+use Sonata\AdminBundle\Datagrid\Pager;
+use Sonata\AdminBundle\Filter\FilterFactoryInterface;
+use Sonata\AdminBundle\Guesser\TypeGuesserInterface;
+use Sonata\DoctrineORMAdminBundle\Builder\DatagridBuilder;
+use Symfony\Component\Form\FormFactoryInterface;
+
+/**
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+final class DatagridBuilderTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var TypeGuesserInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $typeGuesser;
+
+    /**
+     * @var DatagridBuilder
+     */
+    protected $datagridBuilder;
+    /**
+     * @var FormFactoryInterface|\PHPUnit_Framework_MockObject_MockObject
```

Comment:
> Oh so it's for autocompletion… I see.

Not only. Type should be specified IMO.

Mock are special, because both types are acceptable.

> Dunno what the symbol is for that.

I will let it as is for the moment.